src/common/users.ts

```diff
@@ -408,7 +408,7 @@ export const checkAndClearUserStreak = async (
 
   const today = getTodayTz(timezone);
   const day = today.getDay();
-  const difference = differenceInDays(today, lastViewAt);
+  const difference = differenceInDays(today.setHours(0, 0, 0, 0), lastViewAt);
```

Comment:
`setHours` does not return `Date`, it is mutating `today` in place. So better like this?

```suggestion
  today.setHours(0, 0, 0, 0)
  const difference = differenceInDays(today, lastViewAt);
```